odoo/fields.py

```diff
@@ -978,6 +978,9 @@ def determine_value(self, record):
                 self.compute_value(record)
             else:
                 recs = record._in_cache_without(self)
+                if len(recs) > PREFETCH_MAX:
+                    recs = recs[:PREFETCH_MAX] | record
+                recs = recs.with_prefetch(record._prefetch)
```

Comment:
@KangOl I made a test with the use-case above, and it turns out that this line is necessary for performance. It takes 43 queries (sharing `record._prefetch`) instead of 5037 (without sharing `record._prefetch`).